core/cleaner/src/test/java/com/cognifide/aet/cleaner/processors/RemoveArtifactsProcessorTest.java

```diff
@@ -0,0 +1,160 @@
+      .asList("6", "7"));
+
+  private static final Set<String> EMPTY_ARTIFACTS_ID_SET = new HashSet<>();
+
+  private Exchange exchange;
+
+  @Mock
+  private ArtifactsDAO artifactDAO;
+
+  @Mock
+  private CleanerContext cleanerContext;
+
+  @Mock
+  private DBKey dbKey;
+
+  private RemoveArtifactsProcessor removeArtifactsProcessor;
+
+  @Before
+  public void setUp() {
+    when(cleanerContext.isDryRun()).thenReturn(false);
+
+    CamelContext ctx = new DefaultCamelContext();
+    exchange = new DefaultExchange(ctx);
+
+    Message in = exchange.getIn();
+    in.setBody(new ReferencedArtifactsMessageBody("", dbKey));
+    in.setHeader(CleanerContext.KEY_NAME, cleanerContext);
+  }
+
+  @Test
+  public void check_ifRemoveArtifactsWasCalled_expectTrue() throws Exception {
+    when(artifactDAO.getArtifactsId(any(DBKey.class)))
+        .thenReturn(new HashSet<>(ONE_TO_SEVEN_ARTIFACTS_ID_SET));
+    setArtifactsIdToKeep(ONE_TO_FIVE_ARTIFACTS_ID_SET);
+
+    removeArtifactsProcessor = new RemoveArtifactsProcessor(artifactDAO);
+    removeArtifactsProcessor.process(exchange);
+
+    verify(artifactDAO, times(1)).removeArtifacts(any(DBKey.class), any(Set.class));
+    verify(artifactDAO, times(1)).getArtifactsId(any(DBKey.class));
+  }
+
+  @Test
+  public void check_ifRemoveArtifactsWasCalled_expectFalse() throws Exception {
+    when(cleanerContext.isDryRun()).thenReturn(true);
+    when(artifactDAO.getArtifactsId(any(DBKey.class)))
+        .thenReturn(new HashSet<>(ONE_TO_SEVEN_ARTIFACTS_ID_SET));
+    setArtifactsIdToKeep(ONE_TO_FIVE_ARTIFACTS_ID_SET);
+
+    removeArtifactsProcessor = new RemoveArtifactsProcessor(artifactDAO);
+    removeArtifactsProcessor.process(exchange);
+
+    verify(artifactDAO, times(0)).removeArtifacts(any(DBKey.class), any(Set.class));
+    verify(artifactDAO, times(1)).getArtifactsId(any(DBKey.class));
+  }
+
+  @Test
+  public void check_substractArtifactsSets_expectSetOfTwoVariables() throws Exception {
+    when(artifactDAO.getArtifactsId(any(DBKey.class)))
+        .thenReturn(new HashSet<>(ONE_TO_SEVEN_ARTIFACTS_ID_SET));
+    setArtifactsIdToKeep(ONE_TO_FIVE_ARTIFACTS_ID_SET);
+    ReferencedArtifactsMessageBody messageBody = (ReferencedArtifactsMessageBody) exchange.getIn()
+        .getBody();
+    assertEquals(SIX_TO_SEVEN_ARTIFACTS_ID_SET,
+        removeArtifactsProcessor.getArtifactsIdsToRemove(artifactDAO, messageBody));
+  }
+
+  @Test
+  public void check_substractArtifactsSets_expectEmptySet() throws Exception {
+    when(artifactDAO.getArtifactsId(any(DBKey.class)))
+        .thenReturn(new HashSet<>(ONE_TO_FIVE_ARTIFACTS_ID_SET));
+    setArtifactsIdToKeep(ONE_TO_SEVEN_ARTIFACTS_ID_SET);
+    ReferencedArtifactsMessageBody messageBody = (ReferencedArtifactsMessageBody) exchange.getIn()
+        .getBody();
+    assertEquals(EMPTY_ARTIFACTS_ID_SET,
+        removeArtifactsProcessor.getArtifactsIdsToRemove(artifactDAO, messageBody));
```

Comment:
Maybe it would be easier to just check if `removeArtifactsProcessor.getArtifactsIdsToRemove(artifactDAO, messageBody)` is empty?

Reply:
I'm checking it there: https://github.com/Cognifide/aet/pull/336/files#diff-4b628417ba96f36a0a4eec9f4817ba27R144. Here I wanted to check substraction ;)

Reply:
I meant:
`assertTrue(removeArtifactsProcessor.getArtifactsIdsToRemove(artifactDAO, messageBody).isEmpty())` :)